RestaurantReservation.API/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using AuthenticationService;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantReservation.Api.Contracts.Models.Authentication;
+using RestaurantReservation.API.Utilities;
+
+namespace RestaurantReservation.Api.Controllers;
+[Route("api")]
+[ApiController]
+public class AuthenticationController : ControllerBase
+{
+    private readonly IUserValidation _userValidation;
+    private readonly IJwtTokenGenerator _jwtTokenGenerator;
+    private readonly UserManager<ApplicationUser> _userManager;
+    private readonly RoleManager<IdentityRole> _roleManager;
+
+    public AuthenticationController(IUserValidation userValidation,
+        IJwtTokenGenerator jwtTokenGenerator,
+        UserManager<ApplicationUser> userManager,
+        RoleManager<IdentityRole> roleManager)
+    {
+        _userValidation = userValidation;
+        _jwtTokenGenerator = jwtTokenGenerator;
+        _userManager = userManager;
+        _roleManager = roleManager;
+    }
+
+    [HttpPost]
+    [Route("login")]
+    public async Task<IActionResult> Login([FromBody] LoginRequestDto model)
+    {
+        var user = await _userValidation.Validate(model.Username, model.Password);
+        if (user == null)
+        {
+            return Unauthorized();
+        }
+        var token = _jwtTokenGenerator.GenerateToken(user.Value.Item1, user.Value.Item2);
+        if (string.IsNullOrEmpty(token))
+        {
+            return Unauthorized();
+        }
+
+        return Ok(token);
```

Comment:
plz wrap the logic in a service and call it from the controller. always try to keep controllers limited to a single service call, with all processing handled in the service